dask/dataframe/tests/test_arithmetics_reduction.py

```diff
@@ -1477,3 +1473,146 @@ def test_series_agg_with_min_count(method, min_count):
+    assert_eq(ddf["dt1"].std(), pdf["dt1"].std())
+
+    # DataFrame test
+    assert_near_timedeltas(ddf.std().compute(), pdf.std())
+    assert_near_timedeltas(ddf.std().compute(), pdf.std())
+
+
+@pytest.mark.skipif(
+    not PANDAS_GT_120, reason="std() for datetime only added in pandas>=1.2"
+)
+def test_datetime_std_with_larger_dataset():
+    num_rows = 250
+
+    dt1 = pd.Series([pd.NaT] * 15, index=range(15)).append(
+        pd.to_datetime(
+            pd.Series(
+                [
+                    datetime.fromtimestamp(1636426704 + (i * 250000))
+                    for i in range(num_rows - 15)
+                ],
+                index=range(15, 250),
+            )
+        )
+    )
+
+    base_numbers = [
+        (1638290040706793300 + (i * 69527182702409)) for i in range(num_rows)
+    ]
+
+    dt2 = pd.to_datetime(pd.Series(base_numbers))
+    dt3 = pd.to_datetime(pd.Series([int(x * 1.25) for x in base_numbers]))
+    dt4 = pd.to_datetime(pd.Series([int(x * 1.33) for x in base_numbers]))
+
+    pdf = pd.DataFrame(
+        {"dt1": dt1, "dt2": dt2, "dt3": dt3, "dt4": dt4},
+        index=range(250),
+    )
+
+    ddf = dd.from_pandas(pdf, 8)
+
+    # Do NaT checks (not skipped so should be NaTs)
+    assert_eq(ddf[["dt1"]].std(skipna=False), pdf[["dt1"]].std(skipna=False))
+
+    # Same thing but as Series
+    assert ddf["dt1"].std(skipna=False).compute() is pdf["dt1"].std(skipna=False)
+
+    assert_near_timedeltas(ddf.std(skipna=True).compute(), pdf.std(skipna=True))
+    assert_near_timedeltas(ddf.std(skipna=False).compute(), pdf.std(skipna=False))
+
+
+@pytest.mark.skipif(
+    not PANDAS_GT_120, reason="std() for datetime only added in pandas>=1.2"
+)
+def test_datetime_std_across_axis_1():
+    pdf = pd.DataFrame(
+        {
+            "dt1": [
+                datetime.fromtimestamp(1636426704 + (i * 250000)) for i in range(10)
+            ],
+            "dt2": [
+                datetime.fromtimestamp(1636426704 + (i * 217790)) for i in range(10)
+            ],
+        }
+    )
+
+    ddf = dd.from_pandas(pdf, 3)
+
+    assert_eq(ddf.std(axis=1), pdf.std(axis=1))
+
+
+@pytest.mark.skipif(
+    not PANDAS_GT_120, reason="std() for datetime only added in pandas>=1.2"
+)
+@pytest.mark.filterwarnings(
+    "ignore:Dropping of nuisance columns:FutureWarning"
+)  # https://github.com/dask/dask/issues/7714
+def test_datetime_std_across_axis1_null_results():
+    pdf = pd.DataFrame(
+        {
+            "dt1": [
+                datetime.fromtimestamp(1636426704 + (i * 250000)) for i in range(10)
+            ],
+            "dt2": [
+                datetime.fromtimestamp(1636426704 + (i * 217790)) for i in range(10)
+            ],
+            "nums": [i for i in range(10)],
+        }
+    )
+
+    ddf = dd.from_pandas(pdf, 3)
+
+    # Single column always results in NaT
+    assert_eq(ddf[["dt1"]].std(axis=1), pdf[["dt1"]].std(axis=1))
+
+    # Mix of datetimes with other numeric types produces NaNs
```

Comment:
This would be a useful comment in the implementation, I was confused at first why you were creating a manual dataframe